src/components/filter/FilterCreationDialog.tsx

```diff
@@ -115,17 +115,25 @@ export function FilterCreationDialog({
                 );
             }
         },
-        [activeDirectory, snackError, onClose]
+        [activeDirectory, snackError, onClose, filterType]
     );
-
+    const title = useMemo(() => {
```

Comment:
As you wish, but I think this still is a title Id, not a title.
```suggestion
    const titleId = useMemo(() => {
```